### PR TITLE
feat(docs): add authentication guide

### PR DESCRIPTION
**Pull request by brandonroberts**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: app/views/docs/authentication.phtml [hunk 1]
```diff
@@ -0,0 +1,278 @@
+<p>
+    Appwrite provides authentication for many different use cases to fit the needs of developers. 
+    Appwrite manages authentication with a combination of users and sessions.
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:3]: > users and sessions

Maybe user accounts and sessions or accounts and sessions because the API methods say "accounts" and I believe the new UI coming soon will stick with accounts.

File: app/views/docs/authentication.phtml [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+</p>
+
+<p> 
+    Some of the Account API methods are available from the server SDK when you authenticate with JWT. This allows you to perform server-side actions on behalf of your project user.
+</p>
+
+<h2><a href="/docs/authentication#create-an-account" id="create-an-account">Create An Account</a></h2>
+
+<p>
+    A user account in Appwrite is the primary way to access information for a given project. Accounts can be created in many different ways, including email/password, anonymously, using OAuth providers, phone authentication, and more. Users can create and manage authentication sessions through the REST API or client SDKs.
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:20]: >Users can create

I think you mean an Appwrite user. We should call those who develop with Appwrite as developers or Appwrite developers.

This way it's clear they're not a "user" that's created through the account service.

File: app/views/docs/authentication.phtml [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+</p>
+
+<h3><a href="/docs/authentication#email" id="email">Email</a></h3>
+
+<p>
+    Registering an account via email and password is one of the most common ways to sign up for an application. Appwrite provides email and password authentication out of the box. Using one of Appwrite's client SDKs, or the REST APIs directly, you can create an account using an email address and password in your application.
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:26]:
```suggestion
    Registering an account via email and password is one of the most common ways to sign up for an application. Appwrite provides email and password authentication out of the box. Using one of Appwrite's client SDKs or the REST APIs directly, you can create an account using an email address and password in your application.
```

Comment (gewenyu99) [reply app/views/docs/authentication.phtml:26]: According to grammarly :P

File: app/views/docs/authentication.phtml [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+    console.log(response); // Success
+}, function (error) {
+    console.log(error); // Failure
+});
+
+</code></pre>
+</div>
+
+<p>
+    After an account is created, the newly created account must be verified through the account verification route provided by the Appwrite Accounts API, after which the new user can login to their account.
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:57]: >account verification route
Add a link to this route in the API docs

File: app/views/docs/authentication.phtml [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+    console.log(error); // Failure
+});
+
+</code></pre>
+</div>
+
+<h3><a href="/docs/authentication#oauth" id="oauth">OAuth</a></h3>
+
+<p>
+    OAuth is another way to authenticate a user using a multi-step process. When using OAuth to authenticate, the request to authenticate is initiated from the client application, then the user is redirected to an OAuth2 provider to complete the authentication step, and finally, the user is redirected back to the client application if completed successfully. This provides integration with many third-party services that provide their own OAuth integration as a more secure approach than providing a username/password directly.
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:92]:
```suggestion
    OAuth is another way to authenticate a user using a multi-step process. When using OAuth to authenticate, the request to authenticate is initiated from the client application, then the user is redirected to an OAuth2 provider to complete the authentication step, and finally, the user is redirected back to the client application if completed successfully. This provides integration with many third-party services that provide their own OAuth integration as a more secure approach than providing a username and password directly.
```

Comment (gewenyu99) [reply app/views/docs/authentication.phtml:92]: Maybe this will be a more proper way to write in a formal doc like this.

No strong opinions here

File: app/views/docs/authentication.phtml [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+</p>
+
+<p>
+    The example below shows you how to authenticate with OAuth2 using the Web SDK and Amazon's OAuth system.
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:96]: Maybe do GitHub instead of Amazon?

Just a suggestion, I feel like GitHub here's like a competitor or something :P

No strong opinions here.

File: app/views/docs/authentication.phtml [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+;
+
+// Go to OAuth provider login page
+account.createOAuth2Session('amazon');
+
+</code></pre>
+</div>
+
+<p>  
+    If there is already an active session, the new session will be attached to the logged-in account. If there are no active sessions, the server will attempt to look for a user with the same email address as the email received from the OAuth2 provider and attach the new session to the existing user. If no matching user is found - the server will create a new user.
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:118]: Again maybe we should stick to "account" instead of user?

I think of it like this. The user is the person, the account is the digital record for the user. Also because it matches the the createAccount APIs.

File: app/views/docs/authentication.phtml [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+</p>
+
+<h3><a href="/docs/authentication#phone" id="phone">Phone</a></h3>
+
+<p>
+    Phone authentication is done using a two-step authentication process. When using phone authentication, the request to authenticate is initiated from the client application, an SMS is sent to the user with a secret key for creating a session.
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:124]: I think the part where you talk about environment variables required for enabling + setup SMS authentication is missing. Also mention the supported providers?

File: app/views/docs/authentication.phtml [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+</div>
+
+<p>
+    If the provided user ID has not be registered, a new user will be created.
+</p>
+
+<h2><a href="/docs/authentication#login" id="login">Login</a></h2>
+
+<p>
+    Logging in with an email and password is one of the most common ways to login into an application. You can create a session maintained across page reloads and browsing sessions. Appwrite provides email and password authentication out of the box. Using one of Appwrite's client SDKs, or the REST APIs directly, you can manage user login sessions in your application.
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:177]: I feel like a lot of this writing is similar to line 25. Can this be written to be shorter and more concise?

File: app/views/docs/authentication.phtml [hunk 10]
```diff
@@ -0,0 +1,278 @@
...
+    console.log(response); // Success
+}, function (error) {
+    console.log(error); // Failure
+});
+
+</code></pre>
+</div>
+
+<p>
+    After a user is logged in, you can also check whether the user is still logged in after a period of time. The Appwrite Account service provides a get() method that checks whether the current user session is active and returns the user information if successful.
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:208]: hyperlink get()

File: app/views/docs/authentication.phtml [hunk 11]
```diff
@@ -0,0 +1,278 @@
...
+</div>
+
+<p>
+    An authenticated session in appwrite lasts for 1 year and is then automatically destroyed.
+</p>
+
+<h2><a href="/docs/authentication#persistence" id="persistence">Persistence</a></h2>
+
+<p>
+    Appwrite handles the persistence of the session in a consistent way across SDKs. After authenticating with an SDK, the SDK will persist the session so that the user will not need to log in again the next time they open the app. How the SDK persists in the session depends on the SDK.
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:245]: >How the SDK persists in the session depends on the SDK.

I think you mean

"How the session is persisted depends on the SDK."

File: app/views/docs/authentication.phtml [hunk 12]
```diff
@@ -0,0 +1,278 @@
...
+<ul>
+    <li class="margin-bottom-tiny">Web - Uses a session secure cookie and falls back to local storage when a session cookie is not available.</li>
+    <li class="margin-bottom-tiny">Flutter - Uses a session cookie stored in Application Documents through the path_provider package.</li>
+    <li class="margin-bottom-tiny">Apple - Uses a session cookie.</li>
+    <li class="margin-bottom-tiny">Android - Uses a session cookie stored in SharedPreferences.</li>
+</ul>
+
+<h2><a href="/docs/authentication#security" id="security">Security</a></h2>
+
+<p>
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:257]: Link to the permissions page

File: app/views/docs/authentication.phtml [hunk 13]
```diff
@@ -0,0 +1,278 @@
...
+    Security is very important to protect users' data and privacy. Appwrite uses a permissions model coupled with user sessions to ensure users only have access to certain information based on the permissions. With Appwrite services, including databases and storage, access is granted to documents and files at the collection, bucket, document, or file level. This access is consistent across access to these items in relation to document access, file access, and real-time events.
+</p>
+
+<h2><a href="/docs/authentication#best-practices" id="best-practices">Best Practices</a></h2>
+
+<p>
+    Here are some common best practices when handling sessions and keys used for authentication purposes.
+</p>
+
+<ul>
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:267]: This section did not render properly. 

Screen Shot 2022-08-29 at 11 28 37 AM

Comment (gewenyu99) [reply app/views/docs/authentication.phtml:267]: A good time to early dog food the new [contribution guide draft](https://github.com/appwrite/docs/pull/229/files). I don't recall ever seeing unordered lists in our docs, I think we only have ordered list...

File: app/views/docs/authentication.phtml [hunk 14]
```diff
@@ -0,0 +1,278 @@
...
+    <li class="margin-bottom-tiny">Only keep user sessions active as long as needed.</li>
+    <li class="margin-bottom-tiny">Access Appwrite via HTTPS to prevent any redirects from interfering with your requests.</li>
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:269]: would be useful to link to a wiki about this type of attack

File: app/views/docs/authentication.phtml [hunk 15]
```diff
@@ -0,0 +1,278 @@
...
+    <li class="margin-bottom-tiny">Run Appwrite on standard ports to ensure everything works correctly. The complete endpoint is located in the Appwrite Console</li>
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:270]: Is this related to Auth?

File: app/views/docs/authentication.phtml [hunk 16]
```diff
@@ -0,0 +1,278 @@
...
+    <li class="margin-bottom-tiny">Prefer using OAuth2 for authentication with first-party redirects</li>
+    <li class="margin-bottom-tiny">Generate an SSL certification for the Appwrite server</li>
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:272]:
```suggestion
    <li class="margin-bottom-tiny">Generate an SSL certification for the Appwrite instance</li>
```

Comment (gewenyu99) [reply app/views/docs/authentication.phtml:272]: We agreed during writing the terminology guidelines to say "Appwrite Instance" for self-deployed.

File: app/views/docs/authentication.phtml [hunk 17]
```diff
@@ -0,0 +1,278 @@
...
+    <li class="margin-bottom-tiny">Keep API keys private and out of client-side code</li>
+    <li class="margin-bottom-tiny">Only maintain <b>1</b> instance of the Client SDK in your app to avoid conflicting session data</li>
+    <li class="margin-bottom-tiny">When using the realtime service to get users' documents, clean up and resubscribe to the correct resources to maintain the correct app state.</li>
+    <li class="margin-bottom-tiny">Unsubscribing from old resources is also recommended to prevent possible race conditions.</li>
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:276]: If we use punctuation for this list item, lets use a period for all list items :)

**Review by gewenyu99 — CHANGES_REQUESTED** [review]

File: app/views/docs/authentication.phtml [hunk 18]
```diff
@@ -0,0 +1,278 @@
...
+
+<p>
+    While the Users API is integrated from the server-side and operates in an admin scope with access to all your project users, the Account API operates in the scope of the currently logged-in user and usually uses a client-side integration.
+</p>
+
+<p> 
+    Some of the Account API methods are available from the server SDK when you authenticate with JWT. This allows you to perform server-side actions on behalf of your project user.
+</p>
+
+<h2><a href="/docs/authentication#create-an-account" id="create-an-account">Create An Account</a></h2>
```
Comment (gewenyu99) [inline app/views/docs/authentication.phtml:17]: Let's list all the endpoints that can be used to create an account with hyperlinks.

I think it'll be useful if we can find a way for readers that are scanning through to quickly find a list.

File: app/views/docs/authentication.phtml [hunk 19]
```diff
@@ -0,0 +1,281 @@
...
+
+<p>
+    A user account in Appwrite is the primary way to access information for a given project. Accounts can be created in many different ways, including <a href="/docs/client/account#accountCreate">email & password</a>, <a href="/docs/client/account#accountCreateAnonymousSession">anonymous sessions</a>, <a href="/docs/client/account#accountCreateOAuth2Session">OAuth2</a>, <a href="/docs/client/account#accountCreatePhoneSession">phone authentication</a>, and more. Applications can create and manage authentication sessions through the REST API or Client SDKs.
+</p>
+
+<h3><a href="/docs/authentication#email" id="email">Email</a></h3>
+
+<p>
+    Creating an account via email and password is one of the most common ways to sign up for an application. Appwrite provides email and password authentication out of the box. Using one of Appwrite's Client SDKs, or the REST APIs directly, you can create an account using an email address and password in your application.
```
Comment (Meldiron) [inline app/views/docs/authentication.phtml:26]: We could also mention that passwords are hashed with Argon2, a secure password hashing algo.

File: app/views/docs/authentication.phtml [hunk 20]
```diff
@@ -0,0 +1,281 @@
...
+    console.log(response); // Success
+}, function (error) {
+    console.log(error); // Failure
+});
+
+</code></pre>
+</div>
+
+<p>
+    After an account is created, the newly created account must be verified through the <a href="/docs/account#accountCreateVerification">account verification route</a> provided by the Appwrite Accounts API, after which the user can login to their new account.
```
Comment (Meldiron) [inline app/views/docs/authentication.phtml:57]: Is this really the case? I dont think you need verified account to sign in. 

Verification feature can be used with permissions - you can set a collection to be readable only for verified users, for example.

File: app/views/docs/authentication.phtml [hunk 21]
```diff
@@ -0,0 +1,281 @@
...
+</p>
+
+<h3><a href="/docs/authentication#anonymous-user" id="anonymous-user">Anonymous User</a></h3>
+
+<p>
+    Anonymous authentication allows users of your application to create a temporary valid session without creating an account.
```
Comment (Meldiron) [inline app/views/docs/authentication.phtml:63]: The session lasts for 1 year.

File: app/views/docs/authentication.phtml [hunk 22]
```diff
@@ -0,0 +1,281 @@
...
+    console.log(error); // Failure
+});
+
+</code></pre>
+</div>
+
+<h3><a href="/docs/authentication#oauth" id="oauth">OAuth</a></h3>
+
+<p>
+    OAuth is another way to authenticate a user using a multi-step process. When using OAuth to authenticate, the request to authenticate is initiated from the client application, then the user is redirected to an OAuth2 provider to complete the authentication step, and finally, the user is redirected back to the client application if completed successfully. This provides integration with many third-party services that provide their own OAuth integration as a more secure approach than providing a username/password directly.
```
Comment (Meldiron) [inline app/views/docs/authentication.phtml:92]: `if completed successfully`

Also if it fails. Tho you can specify specific URLs for redirect depending if it was successful or not.

File: app/views/docs/authentication.phtml [hunk 23]
```diff
@@ -0,0 +1,281 @@
...
+</p>
+
+<p>
+    The example below shows you how to authenticate with OAuth2 using the Web SDK and Amazon's OAuth system.
```
Comment (Meldiron) [inline app/views/docs/authentication.phtml:96]: We could also mention that they dont need to await it, since it's sync function that triggers redirect in the browser.

File: app/views/docs/authentication.phtml [hunk 24]
```diff
@@ -0,0 +1,281 @@
...
+;
+
+// Go to OAuth provider login page
+account.createOAuth2Session('amazon');
+
+</code></pre>
+</div>
+
+<p>  
+    If there is already an active session, the new session will be attached to the logged-in account. If there are no active sessions, the server will attempt to look for an account with the same email address as the email received from the OAuth2 provider and attach the new session to the existing Account. If no matching account is found - the server will create a new account.
```
Comment (Meldiron) [inline app/views/docs/authentication.phtml:118]: `If there is already an active session, the new session will be attached to the logged-in account.`

This to me sounds like "If you are already logged in with email&password, and you try to login with oauth, these two will now connect to the same account. Not sure if that is true.

File: app/views/docs/authentication.phtml [hunk 25]
```diff
@@ -0,0 +1,281 @@
...
+</p>
+
+<h3><a href="/docs/authentication#phone" id="phone">Phone</a></h3>
+
+<p>
+    Phone authentication is done using a two-step authentication process. When using phone authentication, the request to authenticate is initiated from the client application, an SMS is sent to the user with a secret key for creating a session. 
+</p>
+
+<p> 
+    Phone authentication requires a provider to be setup to send SMS messages. The supported providers are Twillio, Text Magic, and Telesign. After creating an SMS sender with one of these providers, configure these <a href="/docs/environment-variables#phone">environment variables</a> and restart your Appwrite containers.
```
Comment (Meldiron) [inline app/views/docs/authentication.phtml:128]: `The supported providers are Twillio, Text Magic, and Telesign`

Let's not mention the list, I am sure it will grow and we will forget to update it here.

File: app/views/docs/authentication.phtml [hunk 26]
```diff
@@ -0,0 +1,281 @@
...
+</p>
```
Comment (Meldiron) [inline app/views/docs/authentication.phtml:129]: Let's also mention that secret in SMS is.. 6 numbers? I think. Let' double check.

File: app/views/docs/authentication.phtml [hunk 27]
```diff
@@ -0,0 +1,281 @@
...
+const client = new Client();
+
+const account = new Account(client);
+
+client
+    .setEndpoint('https://[HOSTNAME_OR_IP]/v1') // Your API Endpoint
+    .setProject('5df5acd0d48c2') // Your project ID
+;
+
+const promise = account.createPhoneSession('[USER_ID]', '');
```
Comment (Meldiron) [inline app/views/docs/authentication.phtml:147]: What is empty string? If number, lets add some fake phone number

File: app/views/docs/authentication.phtml [hunk 28]
```diff
@@ -0,0 +1,281 @@
...
+    console.log(response); // Success
+}, function (error) {
+    console.log(error); // Failure
+});
+
+</code></pre>
+</div>
+
+<p>
+    If the provided user ID has not be registered, a new account will be created.
```
Comment (Meldiron) [inline app/views/docs/authentication.phtml:175]:
```suggestion
    If the provided user ID has not been registered, a new account will be created.
```

**Review by Meldiron — CHANGES_REQUESTED** [review]

File: app/views/docs/authentication.phtml (hunk 26 above)
Comment (gewenyu99) [reply app/views/docs/authentication.phtml:129]: Yep 6

File: app/views/docs/authentication.phtml (hunk 27 above)
Comment (gewenyu99) [reply app/views/docs/authentication.phtml:147]: Yep

File: app/views/docs/authentication.phtml [hunk 29]
```diff
@@ -0,0 +1,283 @@
+<p>
+    Appwrite provides authentication for many different use cases to fit the needs of developers. 
+    Appwrite manages authentication with a combination of accounts and sessions. 
+    Accounts can be created in many different ways, such as through an anonymous session, email and password, OAuth authentication, magic URLs, and more.
+</p>
+
+<h2><a href="/docs/authentication#account-vs-user-service" id="account-vs-user-service">Account vs User Service</a></h2>
+
+<p>
+    The Account API operates in the scope of the currently logged-in account and usually used in a frontend or mobile app. The Users API is used in backend integrations and operates in an admin scope using an API key with access to all your project users.
```
Comment (christyjacob4) [inline app/views/docs/authentication.phtml:10]:
```suggestion
    The Account API operates in the scope of the currently logged-in account and is usually used in a frontend or mobile app. The Users API is used in backend integrations and operates in an admin scope using an API key with access to all your project users.
```

File: app/views/docs/authentication.phtml [hunk 30]
```diff
@@ -0,0 +1,283 @@
...
+</p>
+
+<p> 
+    Some of the Account API methods are available from Server SDKs when you authenticate with a JWT. This allows you to perform actions on behalf of a project account's user on the backend.
```
Comment (christyjacob4) [inline app/views/docs/authentication.phtml:14]:
```suggestion
    Some of the Account API methods are available from Server SDKs when you authenticate with a JWT. This allows your Server SDK to perform actions on behalf of a user.
```

File: app/views/docs/authentication.phtml [hunk 31]
```diff
@@ -0,0 +1,283 @@
...
+</p>
+
+<h2><a href="/docs/authentication#create-an-account" id="create-an-account">Create An Account</a></h2>
+
+<p>
+    A user account in Appwrite is the primary way to access information for a given project. Accounts can be created in many different ways, including <a href="/docs/client/account#accountCreate">email & password</a>, <a href="/docs/client/account#accountCreateAnonymousSession">anonymous sessions</a>, <a href="/docs/client/account#accountCreateOAuth2Session">OAuth2</a>, <a href="/docs/client/account#accountCreatePhoneSession">phone authentication</a>, and more. Applications can create and manage authentication sessions through the REST API or Client SDKs.
```
Comment (christyjacob4) [inline app/views/docs/authentication.phtml:20]:
```suggestion
    A user account in Appwrite is the primary way to access information for a given project. Accounts can be created in many different ways, including <a href="/docs/client/account#accountCreateEmailSession">email & password</a>, <a href="/docs/client/account#accountCreateAnonymousSession">anonymous sessions</a>, <a href="/docs/client/account#accountCreateOAuth2Session">OAuth2</a>, <a href="/docs/client/account#accountCreatePhoneSession">phone authentication</a>, and more. Applications can create and manage sessions through the REST API or Client SDKs.
```

File: app/views/docs/authentication.phtml [hunk 32]
```diff
@@ -0,0 +1,283 @@
...
+
+promise.then(function (response) {
+    console.log(response); // Success
+}, function (error) {
+    console.log(error); // Failure
+});</code></pre>
+</div>
+
+<p>
+    After an account is created, the newly created account can be verified through the <a href="/docs/account#accountCreateVerification">account verification route</a> provided by the Appwrite Accounts API. The user doesn't need to be verified to log in, but you can restrict resource access to verified users only using permissions.
```
Comment (christyjacob4) [inline app/views/docs/authentication.phtml:59]:
```suggestion
    After an account is created, it can be verified through the <a href="/docs/account#accountCreateVerification">account verification route</a> provided by the Appwrite Accounts API. The user doesn't need to be verified to log in, but you can restrict resource access to verified users only using permissions.
```

File: app/views/docs/authentication.phtml [hunk 33]
```diff
@@ -0,0 +1,283 @@
...
+</p>
+
+<h3><a href="/docs/authentication#anonymous-user" id="anonymous-user">Anonymous User</a></h3>
+
+<p>
+    Anonymous authentication allows users of your application to create a temporary valid session without creating an account. The session will last for a year. If an account is created with an active anonymous session, it will be attached to the existing anonymous session.
```
Comment (christyjacob4) [inline app/views/docs/authentication.phtml:65]:
```suggestion
    Anonymous authentication allows users of your application to create a temporary valid session without creating an account. The session has an expiry of one year. If an account is created while an anonymous session is active, it will be attached to the existing anonymous session.
```

File: app/views/docs/authentication.phtml [hunk 34]
```diff
@@ -0,0 +1,283 @@
...
+    console.log(response); // Success
+}, function (error) {
+    console.log(error); // Failure
+});</code></pre>
+</div>
+
+<h3><a href="/docs/authentication#oauth" id="oauth">OAuth</a></h3>
+
+<p>
+    OAuth is another way to authenticate a user using a multi-step process. When using OAuth to authenticate, the request to authenticate is initiated from the client application, then the user is redirected to an OAuth2 provider to complete the authentication step, and finally, the user is redirected back to the client application. This provides integration with many third-party services that provide their own OAuth integration as a more secure approach than providing a username/password directly.
```
Comment (christyjacob4) [inline app/views/docs/authentication.phtml:92]:
```suggestion
    OAuth is another way to authenticate a user using a multi-step process. When using OAuth to authenticate, the authentication request is initiated from the client application. The user is then redirected to an OAuth2 provider to complete the authentication step, and finally, the user is redirected back to the client application. This provides integration with many third-party services that provide their own OAuth integration as a more secure approach than providing a username/password directly.
```

File: app/views/docs/authentication.phtml [hunk 35]
```diff
@@ -0,0 +1,283 @@
...
+    .setEndpoint('https://[HOSTNAME_OR_IP]/v1') // Your API Endpoint
+    .setProject('5df5acd0d48c2') // Your project ID
+;
+
+// Go to OAuth provider login page
+account.createOAuth2Session('amazon');</code></pre>
+</div>
+
+<p>
+    Notice, the function call for <b>createOAuth2Session</b> is not asynchronous. No need to use <b>await<b>.
```
Comment (christyjacob4) [inline app/views/docs/authentication.phtml:116]:
```suggestion
    Notice, the function call for <b>createOAuth2Session</b> is synchronous and hence, there is no need to use <b>await<b>.
```

File: app/views/docs/authentication.phtml [hunk 36]
```diff
@@ -0,0 +1,283 @@
...
+</p>
+
+<p>  
+    If there is already an active anonymous session, the new session will be attached to the logged-in account. If there are no active sessions, the server will attempt to look for an account with the same email address as the email received from the OAuth2 provider and attach the new session to the existing account. If no matching account is found - the server will create a new account.
```
Comment (christyjacob4) [inline app/views/docs/authentication.phtml:120]:
```suggestion
    If there is already an active anonymous session, the new session will be attached to it. If there are no active sessions, the server will attempt to look for an account with the same email address as the email received from the OAuth2 provider and attach the new session to the existing account. If no matching account is found - the server will create a new account.
```

File: app/views/docs/authentication.phtml [hunk 37]
```diff
@@ -0,0 +1,283 @@
...
+<p>
+    Phone authentication is done using a two-step authentication process. When using phone authentication, the request to authenticate is initiated from the client application, an SMS is sent to the user with a secret key for creating a session. 
+</p>
+
+<p> 
+    Phone authentication requires you to set up an SMS provider to send SMS messages. You will need to configure these <a href="/docs/environment-variables#phone">environment variables</a> and restart your Appwrite containers before you can use phone authentication.
+</p>
+
+<p>
+    The example below shows you how to initiate a phone authentication using the Web SDK.
```
Comment (christyjacob4) [inline app/views/docs/authentication.phtml:134]:
```suggestion
    The example below shows you how to initiate a phone authentication request using the Web SDK.
```

File: app/views/docs/authentication.phtml [hunk 38]
```diff
@@ -0,0 +1,283 @@
...
+    console.log(response); // Success
+}, function (error) {
+    console.log(error); // Failure
+});
+
+</code></pre>
+</div>
+
+<p>
+    After initiation, the returned username and secret are used to confirm the user. The secret will be a 6 digit number in the SMS message sent to the user.
```
Comment (christyjacob4) [inline app/views/docs/authentication.phtml:161]:
```suggestion
    After initiation, the returned user ID and secret are used to confirm the user. The secret will be a 6 digit number in the SMS message sent to the user.
```

File: app/views/docs/authentication.phtml [hunk 39]
```diff
@@ -0,0 +1,283 @@
...
+    console.log(response); // Success
+}, function (error) {
+    console.log(error); // Failure
+});
+
+</code></pre>
+</div>
+
+<p>
+    If the provided user ID has not been registered, a new account will be created.
```
Comment (christyjacob4) [inline app/views/docs/authentication.phtml:177]: This is incorrect. A new user is not created here
https://github.com/appwrite/appwrite/blob/bf8e113e703084834547ddebe0747108891907f7/app/controllers/api/account.php#L2356

File: app/views/docs/authentication.phtml [hunk 40]
```diff
@@ -0,0 +1,283 @@
...
+const client = new Client();
+
+const account = new Account(client);
+
+client
+    .setEndpoint('https://[HOSTNAME_OR_IP]/v1') // Your API Endpoint
+    .setProject('5df5acd0d48c2') // Your project ID
+;
+
+const promise = account.getSession('current');
```
Comment (christyjacob4) [inline app/views/docs/authentication.phtml:233]: Why do we need to use the `getSession` here ? 
Can't we just use this 
https://appwrite.io/docs/client/account?sdk=web-default#accountGet

File: app/views/docs/authentication.phtml [hunk 41]
```diff
@@ -0,0 +1,283 @@
...
+    console.log(response); // Success
+}, function (error) {
+    console.log(error); // Failure
+});
+
+</code></pre>
+</div>
+
+<p>
+    An authenticated session in appwrite lasts for 1 year and is then automatically destroyed.
```
Comment (christyjacob4) [inline app/views/docs/authentication.phtml:245]:
```suggestion
    An authenticated session in Appwrite lasts for 1 year and is then automatically expired.
```

File: app/views/docs/authentication.phtml [hunk 42]
```diff
@@ -0,0 +1,283 @@
...
+</p>
+
+<h2><a href="/docs/authentication#persistence" id="persistence">Persistence</a></h2>
+
+<p>
+    Appwrite handles the persistence of the session in a consistent way across SDKs. After authenticating with an SDK, the SDK will persist the session so that the user will not need to log in again the next time they open the app. How the session is persisted depends on the SDK.
```
Comment (christyjacob4) [inline app/views/docs/authentication.phtml:251]:
```suggestion
    Appwrite handles the persistence of the session in a consistent way across SDKs. After authenticating with an SDK, the SDK will persist the session so that the user will not need to log in again the next time they open the app. The mechanism for persistence depends on the SDK.
```

File: app/views/docs/authentication.phtml [hunk 43]
```diff
@@ -0,0 +1,283 @@
...
+<p>
+    Here are some common best practices when handling sessions and keys used for authentication purposes.
+</p>
+
+<ol>
+    <li class="margin-bottom-tiny">Only keep user sessions active as long as needed.</li>
+    <li class="margin-bottom-tiny">Access Appwrite via HTTPS to prevent any redirects from interfering with your requests.</li>
+    <li class="margin-bottom-tiny">Prefer using OAuth2 for authentication with first-party redirects.</li>
+    <li class="margin-bottom-tiny">Generate an SSL certification for the Appwrite instance.</li>
+    <li class="margin-bottom-tiny">Keep API keys private and out of client-side code.</li>
```
Comment (christyjacob4) [inline app/views/docs/authentication.phtml:278]: We can add a point about not giving an API key more scopes than it actually needs.

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Please address the comments 👍

**Review by christyjacob4 — APPROVED** [review]
